### DISCUSSION TITLE
DataGrid edit on double click

### DISCUSSION THREAD

**Discussion by dockinkong** [Megabit/Blazorise#3131] [General]
Hello,

Is there a way to call edit mode on row when double click on it in the DataGrid please?

**Comment by stsrki** [comment]
You can use the `RowDoubleClicked` event to listen for mouse double click. Then you can just call `await dataGrid.Edit( e.Item )` to switch into edit mode.

**Reply by dockinkong** [reply]
Oh yes, perfect :). Thanks !

Is there a way to avoid row to be selected/unselected when double clicking?

**Reply by stsrki** [reply]
Not sure. @David-Moreira do you know the answer?

**Reply by David-Moreira** [reply]
Hello,
With our current click detection, no, don't think so. But I believe you can tinker with the `Datagrid` to achieve a similar behaviour.

You could: 
1. Disable RowSelectable & use Multiple Selection instead to handle selection.
https://blazorise.com/docs/extensions/datagrid/selection | RowSelectable & Multiple Selection
2. Disable RowSelectable and In theory you should be able to override the selection and do something like : Bind to RowClick and apply a very tiny timer that checks if DoubleClick was pressed within that time frame, if it wasn't then you can use `DataGrid.Select(Item)`
3. Disable RowSelectable and create a new column with a button that selects the row  `DataGrid.Select(Item)`. (This keeps the DataGrid in Single Selection Mode)
